service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderControllerSubmittedEvent.java

```diff
@@ -111,35 +132,191 @@ void shouldPublishPopulateStandardDirectionsEventIfAnAdditionalSDOHasBeenRemoved
             eq(12345L),
             eq("populateSDO"),
             anyMap());
+
+        verify(notificationClient).sendEmail(
+            eq(SDO_REMOVAL_NOTIFICATION_TEMPLATE),
+            eq(GATEKEEPER_EMAIL_ADDRESS),
+            eqJson(expectedOrderRemovalTemplateParameters()),
+            eq(NOTIFICATION_REFERENCE));
     }
 
     @Test
-    void shouldNotPublishPopulateStandardDirectionsEventIfASDOHasNotBeenRemoved() {
+    void shouldNotPublishPopulateSDOAndSDORemovedEventsIfASDOHasNotBeenRemoved() {
         StandardDirectionOrder previousSDO = StandardDirectionOrder.builder().build();
 
-        List<Element<StandardDirectionOrder>> hiddenSDOs = new ArrayList<>();
-        hiddenSDOs.add(element(previousSDO));
+        List<Element<StandardDirectionOrder>> hiddenSDOs = singletonList(element(previousSDO));
+        List<Element<CaseManagementOrder>> hiddenCMOs = singletonList(element(CaseManagementOrder.builder().build()));
+
+        CaseDetails caseDetails = caseDetailsWithRemovableOrders(hiddenCMOs, hiddenSDOs, emptyList());
+        CaseDetails caseDetailsBefore = caseDetailsWithRemovableOrders(emptyList(), hiddenSDOs, emptyList());
 
-        CaseDetails caseDetails = CaseDetails.builder()
-            .data(Map.of(
-                "hiddenStandardDirectionOrders", hiddenSDOs,
-                "hiddenCaseManagementOrders", List.of(element(CaseManagementOrder.builder().build()))
-            ))
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .caseDetailsBefore(caseDetailsBefore)
             .build();
 
-        CaseDetails caseDetailsBefore = CaseDetails.builder()
-            .data(Map.of(
-                "hiddenStandardDirectionOrders", hiddenSDOs
-            ))
+        postSubmittedEvent(callbackRequest);
+
+        verifyNoMoreInteractions(coreCaseDataService);
+        checkThat(() -> verify(notificationClient, never())
+            .sendEmail(eq(SDO_REMOVAL_NOTIFICATION_TEMPLATE), any(), any(), any()));
+    }
+
+    @Test
+    void shouldNotSendEmailNotificationsIfABlankOrderIsRemoved() {
```

Comment:
Blank order is a generatedOrderType. Investigate wether you can include it in the enum source for below test.

Reply:
included generated order to the parameterized test and removed `shouldNotSendEmailNotificationsIfABlankOrderIsRemoved` test